packages/scaffold/src/views/w3m-wallet-send-preview-view/index.ts

```diff
@@ -114,11 +129,64 @@ export class W3mWalletSendPreviewView extends LitElement {
     return null
   }
 
-  private onSendClick() {
-    RouterController.reset('Account')
-    setTimeout(() => {
-      SendController.resetSend()
-    }, 200)
+  private async onSendClick() {
+    if (this.token?.address && this.sendTokenAmount) {
+      RouterController.pushTransactionStack({
+        view: 'Account',
+        goBack: false
+      })
+
+      const amount = ConnectionController.parseUnits(
+        this.sendTokenAmount.toString(),
+        Number(this.token.quantity.decimals)
+      )
+
+      try {
+        if (
+          AccountController.state.address &&
+          this.sendTokenAmount &&
+          this.receiverAddress &&
+          this.token.address
+        ) {
+          await ConnectionController.writeContract({
+            fromAddress: AccountController.state.address as `0x${string}`,
+            tokenAddress: CoreHelperUtil.extractEthereumAddress(
+              this.token.address
+            ) as `0x${string}`,
+            receiverAddress: this.receiverAddress as `0x${string}`,
+            tokenAmount: amount
+          })
+          RouterController.replace('Account')
+          SnackController.showSuccess('Transaction Successful')
+        }
+      } catch (error) {
+        SnackController.showError('Something went wrong...')
+      }
+    } else if (this.receiverAddress && this.sendTokenAmount && this.gasPrice) {
+      RouterController.pushTransactionStack({
+        view: null,
+        goBack: true
+      })
+
+      const to = this.receiverAddress as `0x${string}`
+      const address = AccountController.state.address as `0x${string}`
+      const value = ConnectionController.parseUnits(this.sendTokenAmount.toString(), 18)
+      const data = '0x'
```

Comment:
to my point before for a generic `writeContract` you could encode all of this call into callData here.

Reply:
This is for native tokens for a `sendTransaction`, not a `writeContract`